RestrictAnonymousAccess.php

```diff
@@ -129,6 +129,8 @@ protected function isAllowedRequest()
     {
         $allowedReferrers = $this->getAllowedReferrers();
         $referrer = Common::sanitizeInputValues(@$_SERVER['HTTP_REFERER']);
+        // Fix for deprecation error in PHP8
+        if(is_null($referrer)) return false;
```

Comment:
```suggestion
        if(empty($referrer)) return false;
```
In questo modo includiamo eventualmente anche il caso della stringa vuota e ritorniamo subito al chiamante.